openupgradelib/openupgrade_merge_records.py

```diff
@@ -347,24 +348,27 @@ def _adjust_merged_values_orm(env, model_name, record_ids, target_record_id,
             elif op == 'or':
                 vals[field.name] = functools.reduce(lambda x, y: x | y, l)
         elif field.type in ('date', 'datetime'):
+            l = filter(lambda x: x is not False, l)
```

Comment:
You can check before if the `op` is going to apply any operation for performance reason